sdk/inc/azure/platform/az_mqtt_mosquitto.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Defines MQTT constructs for Mosquitto.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_MQTT_MOSQUITTO_H
+#define _az_MQTT_MOSQUITTO_H
+
+#include "mosquitto.h"
+#include <azure/core/internal/az_mqtt_internal.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+typedef struct
+{
+  az_mqtt_options_common platform_options;
+  /**
+   * The CA Trusted Roots span interpretable by the underlying MQTT implementation.
```

Comment:
We should also add that we recommend applications avoid configuring this and instead relying on the default OpenSSL default `ca-certificates` package.
Looks like what we need to do according to @vaavva 's tests:

> I could get this to work by setting MOSQ_OPT_TLS_USE_OS_CERTS and passing in a dummy value in ca_path that would then be ignored (ex. 's')

Reply:
Ref: https://github.com/eclipse/mosquitto/issues/1824

Reply:
Added changes to rely on OS certificates